woocommerce/woocommerce-ios
# Login: Show instruction to connect to a WP.com site from the account mismatch error screen

Closes: #7766 


### Description

This PR adds a new alert to assist users in accessing the app when they use accounts that don't have access to their WP.com sites. Since users cannot connect to WP.com sites by themselves, the suggestion is to contact the site owner for an invitation to the site as a shop manager or administrator to be able to use the app.

Previously in #7748, an activity indicator is displayed on the account mismatch screen while fetching the site info since we intended not to show the Connect Jetpack button if the entered site is a WP.com. In this PR, the primary button should be displayed for all cases, so the activity indicator is removed, and the button should show the loading state while the site info is being fetched.

The primary button on the account mismatch screen is also renamed to "Connect to the site" since the previous title "Connect Jetpack" doesn't match this case.

This PR also fixes an issue when the user is stuck on the Jetpack plans page when handling Jetpack connection for a self-hosted site. For some reason, the delegate method `decidePolicy` is not triggered when the web view redirects to the Jetpack plans page, so the solution is to catch the redirect in the `handleRedirect` method.

### Testing instructions

- Log out of the app if needed, and select Enter your site address.
- Enter the address of a WP.com store.
- Proceed to log in with a WP.com account that doesn't have access to the above store.
- After the login succeeds, notice that the account mismatch screen is displayed. The primary button should be present and shows the loading state initially.
- After the loading is done, tap on the primary button. Notice that a new alert is displayed with the suggestion on how to connect to the store.

Please feel free to repeat the above steps with a self-hosted site. Tapping on the primary button should show the site credential login flow instead.

### Screenshots

Logging in to a WP.com site:

https://user-images.githubusercontent.com/5533851/192456853-0d7f4770-9415-4ca3-bc71-f44ed3f1a65b.mp4

Logging in to a self-hosted site:


https://user-images.githubusercontent.com/5533851/192456947-4db86fca-a3a9-4adc-8858-ca6c652abf36.mp4



---
